File: content/snippets/scala2/language_basics/maps_and_sets.scala
object MapsAndSetsBasics {
  def main(args: Array[String]): Unit = {
    val counts = scala.collection.mutable.Map("apple" -> 2, "banana" -> 1)
    counts("apple") = counts("apple") + 3

    val tags = scala.collection.mutable.Set("fruit", "food")
    tags += "fresh"

    println(s"apple count: ${counts("apple")}")
    println(s"all map entries: ${counts.toSeq.sortBy(_._1)}")
    println(s"all set entries: ${tags.toSeq.sorted}")
  }
}

File: content/snippets/typescript/language_basics/maps_and_sets.ts
#!/usr/bin/env ts-node

import { setImmediate } from "node:timers";

function main(): void {
  const counts = new Map<string, number>([
    ["apple", 2],
    ["banana", 1],
  ]);
  counts.set("apple", (counts.get("apple") ?? 0) + 3);

  const tags = new Set<string>(["fruit", "food"]);
  tags.add("fresh");

  console.log("apple count:", counts.get("apple"));
  console.log("all map entries:", Array.from(counts.entries()));
  console.log("all set entries:", Array.from(tags.values()));
}

main();
setImmediate(() => {});
